hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/offset/PendingOffsets.java

```diff
@@ -72,6 +76,8 @@ public Map<SubscriptionPartitionOffset, MessageState> getOffsetsSnapshotAndRelea
             if (entry.getValue() == MessageState.PROCESSED) {
                 slots.remove(entry.getKey());
                 permitsReleased++;
+            } else if (entry.getValue() == MessageState.FILTERED) {
```

Comment:
Maybe it would be convenient to have a comment that we only increase semaphore for sending messages, and omitting increasing for filtered, but in any case we put a slot for both of the messages. I just had some problems with understanding how there can be slots in the queue but we didn't increased semaphore for them, thus the comment would help.

Reply:
I have changed the approach and now filtered messages acquires semaphore, please take a look now 😄